File: algorithms/1827.minimum-operations-to-make-the-array-increasing/solution.rb
# @param {Integer[]} nums
# @return {Integer}
def min_operations(nums)
  count = prev = 0
  nums.each do |num|
    unless num > prev
      count += (prev + 1) - num
      prev += 1
    else
      prev = num
    end
  end
  count
end

File: algorithms/1827.minimum-operations-to-make-the-array-increasing/solution.rs
impl Solution {
    pub fn min_operations(nums: Vec<i32>) -> i32 {
        let mut count: i32 = 0;
        let mut prev: i32 = 0;
        for num in nums {
            if prev >= num {
                count += (prev + 1) - num;
                prev += 1
            } else {
                prev = num
            }
        }
        count
    }
}
